### DISCUSSION TITLE
Problem using select_features function in EEG data

### DISCUSSION THREAD

**Discussion by basaravia** [blue-yonder/tsfresh#890] [Q&A] [answered]
Hi guys, we are very grateful with your work with tsfresh, We are using EEG signals in a dataframe that has the columns: index (default integer index from pandas), “time” (miliseconds) ,”F3” ,”F4” ,”O1” ,”O2” (electrode signals in microvolts) and an “id” (1 to 5 label). We applied the function “extract_features”, and we obtain a 200x3149 data frame.

The problem appears to be when we apply the “impute” and “select_features” function the output is only a dataframe 200x33. 

Doy you have any suggestions to get more than this 33 selected features?

**Comment by kempa-liehr** [comment, marked as answer]
Hi,
There are several options:
- You could increase the number of time-series samples, either by doing more experiments, or by "chunking" the time-series from your 200 experiments into smaller chunks (or windows), which would all have the same label than the original time-series. Make sure, that you also keep the ID of your original time-series as group-ID, such that you can use leave-K-group-out cross-validation in your machine learning pipeline.
- You could design or "engineer" new time-series, e.g. by computing the time derivative of input time-series F3, F4, O1, and O2.
- If your time-series series are synchronised, you could also compute the difference between signals, eg. F3-F4 and O1-O2 and generate time-series features from these difference time-series ([time-series engineering](https://dx.doi.org/10.1007/978-981-15-3651-9_20)).
- Imputation might also cause problems, such that you could decide to [remove missing values from your time-series](https://dx.doi.org/10.1016/j.ascom.2021.100460) before extracting time-series features. If your time-series are sufficiently long, this approach should generate better results compared to imputation.

Cheers,
Andreas
